AraEvent/AraEventCalibrator.cxx

```
@@ -2145,14 +2148,19 @@ Int_t AraEventCalibrator::numberOfPedestalValsInFile(char *fileName){
     gzFile inPed = gzopen(fileName, "r");
     if(inPed){
         std::vector<char> buffer(6000000);
-        int nRead = gzread(inPed, &buffer[0], sizeof(char)*6000000);
-        std::string string_buffer(buffer.begin(), buffer.end()); // convert the vector of chars to a string
+        int nRead = gzread(inPed, buffer.data(), buffer.size());
+        if(nRead < 0) {
+            throw std::runtime_error("Error reading ped file: "+std::string(fileName));
+        }
+        std::string string_buffer(buffer.data(), nRead); // convert the vector of chars to a string
         std::stringstream ss(string_buffer); // and then convert the string to a stringstream
 
         std::string dummy;
         while( ss >> dummy >> dummy >> dummy){
             for(int samp=0; samp < SAMPLES_PER_BLOCK; samp++){
-                ss >> dummy;
+                if(!(ss >> dummy)){
```

Do we break, or should we throw an error too?

---

hmm I'm not sure. Probably just a break? The break will ensure the count is accurate and then the existing if statement will catch if the count isn't what's expected? So I think this is okay since under the current setup its not this function's job to catch errors in the file size. But let me know if you'd rather it just crash.

---

Do we have some other check to make sure it gets the number of peds right?

---

There's the one that I changed the error message for.

---

Yeah, ok